protocol/testing/e2e/gov/wind_down_market_test.go

```diff
@@ -0,0 +1,287 @@
+package gov_test
+
+import (
+	"testing"
+
+	"github.com/cometbft/cometbft/types"
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/liquidation/api"
+	"github.com/dydxprotocol/v4-chain/protocol/indexer"
+	indexerevents "github.com/dydxprotocol/v4-chain/protocol/indexer/events"
+	"github.com/dydxprotocol/v4-chain/protocol/indexer/indexer_manager"
+	"github.com/dydxprotocol/v4-chain/protocol/indexer/msgsender"
+	indexershared "github.com/dydxprotocol/v4-chain/protocol/indexer/shared"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	prices "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+func TestWindDownMarketProposal(t *testing.T) {
+	tests := map[string]struct {
+		subaccounts    []satypes.Subaccount
+		statefulOrders []clobtypes.MsgPlaceOrder
+
+		expectedSubaccounts []satypes.Subaccount
+	}{
```

Comment:
can we add a few more test cases 
- trading is disabled (new messages rejected)
- trading is disabled during PCS replay (not sure if there is a good way to test this with e2e framework)
- liquidation is disabled

Reply:
i think some of these tests are things that I can break out into a different e2e test which doesn't need to include the passing of the gov proposal in the test suite. The purpose of this test suite in particular was to verify things that happen as side effects of the proposal passing (open orders cancelled, open positions closed). Will break these out